First Pass on Docs Redesign

toledoal opened the pull request:
Hello Peli!

This is the first pass of the re-design of Docs. Please let me know if there is something you would like in a different way.

1. We use the primary color as accent color to be in sync with the editor.
2. Added some html markup elements to match de design (mainly app header)
3. Modify the docs.less file
4. We added the "Lato" fonts. Would you like this to be placed locally in a folder "docs/static/fonts" or is it ok to reference it thru an url
5. We would like to add new interaction features to the sidebar but it is created dynamically by docsrender.ts. We would like to include icon markup and other classes. 
6. Currently we are working more on the mobile experience and localization switch

Pending To Do's

- [x] Set media queries width size for sidebar (Small phones)
- [x] Place the Hamburger menu to the right and leave "logo" fixed to the left

pelikhan commented on docfiles/appheader.html:
```diff
@@ -0,0 +1,15 @@
+<div id="app-header" class="ui main menu @tocclass@ borderless hideprint">
```
Rename to “docsheader”

pelikhan commented on docfiles/docs.html:
```diff
@@ -17,6 +17,7 @@
 		@accMenu@
 	</div>
 
+	<!-- @include appheader.html-->
```
Docs header

pelikhan commented on docs/static/Microsoft_Makecode.svg:
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
We have plenty of icons under /static/icons . Is this one new?

toledoal replied:
Yes, this is a new one. But now that I see it I think its place in under the docs/static/logo folder. We will move it there.

pelikhan replied:
Stick to “-“ instead of _ in files names. Avoid capital letters too (Linux vs windows quirks)

pelikhan commented on docfiles/header.html:
```diff
@@ -1,5 +1,2 @@
-    </div>
-</div>
+<div class="ui one padded column grid hideprint hidemobile">
+       <!-- @breadcrumb@ Check before commit--> </div>
```
???

toledoal replied:
yes, this shouldn't be here.

pelikhan commented on theme/docs.less:
```diff
@@ -28,9 +29,44 @@
 @docsCardBorderColor: #e9eef2;
 @docsCardHoverBorderColor: #1dacf4;
 
+#app-header {
```
docs header

pelikhan commented on theme/docs.less:
```diff
@@ -185,8 +265,147 @@
+        content: '';
+        display: inline-block;
+        -webkit-mask: url(static/folder-menu.svg) no-repeat;
+        mask: url(static/folder-menu.svg) no-repeat;
```
Use data-uri to inline image

pelikhan commented on theme/themes/pxt/globals/site.variables:
```diff
@@ -33,13 +34,13 @@
 --------------------*/
 
-@importGoogleFonts: false;
+@importGoogleFonts: true;
```
Revert this.

pelikhan commented on theme/themes/pxt/globals/site.variables:
```diff
@@ -33,13 +34,13 @@
 
-@docsHeaderFont: "Helvetica Neue", Helvetica, Arial, sans-serif;
-@docsPageFont: "Helvetica Neue", Helvetica, Arial, sans-serif;
+@docsHeaderFont: Lato,"Helvetica Neue", Helvetica, Arial, sans-serif;
```
Where is that font coming from? We probably need to file an oss request for it.

toledoal replied:
Yes, this is the new "Lato" font we were using from google font. We could donwload the web font and store it in docs/static/fonts, is that ok? That would solve us the issue of getting the font from google

pelikhan replied:
What is the license for that font? We need to file an OSS request for it.

pelikhan replied:
Does it support non-European scripts?

pelikhan replied:
Let me look into adding to pxt myself. There is some administrative task attached to this. Maybe remove it from this PR and make it a separate PR?

pelikhan replied:
Let me look into adding to pxt myself. There is some administrative task attached to this. Maybe remove it from this PR and make it a separate PR?

pelikhan replied:
Right we probably need a fallback font.

pelikhan commented:
Looks good overall.
- could you add screenshots of the current changes? For pxt docs, Microbit docs and adafruit docs.
- do not enable google fonts. If we need a new font, we should go through the regular OSS process to host on our back end.
- avoid calling the header “app something” as it is confusing wrt to our Windows/Web apps . Let’s call it “docs-header”

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/pxt?pullRequest=5313) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/pxt?pullRequest=5313) before we can accept your contribution.<br/><br/>:x: toledoal [sign now](https://cla.opensource.microsoft.com/Microsoft/pxt?pullRequest=5313)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/pxt?pullRequest=5313) it.</sub>

pelikhan commented on theme/docs.less:
```diff
@@ -269,8 +274,8 @@ height:100%;
-        -webkit-mask: url(static/folder-menu.svg) no-repeat;
-        mask: url(static/folder-menu.svg) no-repeat;
+        -webkit-mask: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMTZweCIgaGVpZ2h0PSIxMHB4IiB2aWV3Qm94PSIwIDAgMTYgMTAiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayI+CiAgICA8IS0tIEdlbmVyYXRvcjogU2tldGNoIDUzLjIgKDcyNjQzKSAtIGh0dHBzOi8vc2tldGNoYXBwLmNvbSAtLT4KICAgIDx0aXR ...
+        mask: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMTZweCIgaGVpZ2h0PSIxMHB4IiB2aWV3Qm94PSIwIDAgMTYgMTAiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayI+CiAgICA8IS0tIEdlbmVyYXRvcjogU2tldGNoIDUzLjIgKDcyNjQzKSAtIGh0dHBzOi8vc2tldGNoYXBwLmNvbSAtLT4KICAgIDx0aXRsZT5JY29 ...
```
LESS has a macro for this. I think it is data-uri

toledoal replied:
Ahh, yes. Got it!

pelikhan commented:
Please add screenshots.

toledoal commented:
Here are some screenshots. One thing we noticed until going to the microbit target, was the nested accordions. We are currently styling those. Is it ok if we add those changes to this PR? Thanks.

We use @primaryColor as the base accent color for Docs as well.

![image](https://user-images.githubusercontent.com/704682/53900353-00db6000-4002-11e9-97ee-c3953b4a2aec.png)

pelikhan commented:
Yep accordions need a bit of love

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Alejandro Toledo Martinez <notifications@github.com>
Sent: Wednesday, March 6, 2019 9:24:18 AM
To: Microsoft/pxt
Cc: Peli de Halleux; Comment
Subject: Re: [Microsoft/pxt] First Pass on Docs Redesign (#5313)


Here are some screenshots. One thing we noticed until going to the microbit target, was the nested accordions. We are currently styling those. Is it ok if we add those changes to this PR? Thanks.

We use @primarycolor<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fprimarycolor&data=02%7C01%7Cjhalleux%40microsoft.com%7C9d17d2f6d0c74c5bab4108d6a25890d7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636874898601664982&sdata=5pjawMNzkRGKMH5hr7QDCT7jIvWLHN0SN%2BWcBmfRL6M%3D&reserved=0> as the base accent color for Docs as well.

[image]<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F704682%2F53900353-00db6000-4002-11e9-97ee-c3953b4a2aec.png&data=02%7C01%7Cjhalleux%40microsoft.com%7C9d17d2f6d0c74c5bab4108d6a25890d7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636874898601664982&sdata=jWMeZzuHRgCCzQzj%2BtmB2oAoXRi8XAKVVvHRQgOiJLQ%3D&reserved=0>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2Fpxt%2Fpull%2F5313%23issuecomment-470197831&data=02%7C01%7Cjhalleux%40microsoft.com%7C9d17d2f6d0c74c5bab4108d6a25890d7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636874898601664982&sdata=qQtwct8IwOJrx1HSPBPYZnwmmoUd%2FdSj78kIlKbxe8s%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAD-4KUX8ytnZQX8cJT2jExheUogh8NHSks5vT_nCgaJpZM4be8sH&data=02%7C01%7Cjhalleux%40microsoft.com%7C9d17d2f6d0c74c5bab4108d6a25890d7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636874898601821229&sdata=16r117hKiHuqobfhV2xe%2BRyhysNHN%2Ff3QPGxcsqbaJQ%3D&reserved=0>.

abchatra commented:
Looks pretty slick @toledoal

toledoal commented:
New screenshot of Arcade: 
![image](https://user-images.githubusercontent.com/704682/54229853-c45ca800-44ca-11e9-82f1-df86cf2a144f.png)

toledoal commented:
We have the following To Do items: 
1. Set media queries width size for sidebar (Small phones)
2. Place the Hamburger menu to the right and leave "logo" fixed to the left

pelikhan commented:
@toledoal use "- [ ] blabla" syntax in the workitem description and github will track todos for you.

pelikhan commented:
github has this neat "resolve conversation" when you address a comment. it help keep track of things.

pelikhan commented on docfiles/docs.html:
```diff
@@ -7,6 +7,7 @@
 	<meta name="Description" content="@description@" />
 
 	<!-- @include meta.html -->
+
```
not needed

pelikhan commented:
You need to test the editor as well. The Arcade banner is now purple too.
![image](https://user-images.githubusercontent.com/4175913/55130255-67bad900-50d7-11e9-8640-e59ed920e781.png)

pelikhan commented:
pxt docs looking very slick

toledoal commented on docfiles/docs.js:
```diff
@@ -257,8 +258,19 @@ function renderSnippets() {
 
+function responsiveResize(){
+    // Add Responsive Adjustments depending on screen size 
+    $( window ).resize(function() {
```
got it. Will refactor this to use pure vanilla, or are we planning to use something different in docs? This will also have to do with issue https://github.com/Microsoft/pxt/issues/5347 regarding lang picker. We can always just use vanilla js :)

toledoal replied:
I was checking that Semantic UI depends on jquery 😟

I found this https://github.com/morganbarrett/Vanilla-Semantic-UI that tries to remove that dependency. 

I think in this case (docs) we would need to talk more about this. What do you think?

toledoal replied:
Perfect, I'll try to not use jquery if possible 😄

toledoal commented on docfiles/footer.html:
```diff
@@ -1,21 +1,5 @@
 <footer class="ui vertical footer segment hideprint" aria-hidden="false">
-    <div class="ui center aligned container">
-        <div class="ui horizontal link list">
-            <a class="item" href="https://twitter.com/intent/follow?screen_name=MsMakeCode" title="Follow us on Twitter" target="_blank"
```
Ah yes, per our conversations on the design the agreement was to remove these links from the footer. We can always put those back with a different style and/or move those to a different place.

pelikhan commented:
We still have some heavy dependency on jquery for docs rendering. Try to avoid it but don’t try to remove it completely.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Alejandro Toledo Martinez <notifications@github.com>
Sent: Thursday, March 28, 2019 4:54:25 PM
To: Microsoft/pxt
Cc: Peli de Halleux; Comment
Subject: Re: [Microsoft/pxt] First Pass on Docs Redesign (#5313)


@toledoal commented on this pull request.

________________________________

In docfiles/docs.js<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2Fpxt%2Fpull%2F5313%23discussion_r270241070&data=02%7C01%7Cjhalleux%40microsoft.com%7Ca8d6f4d9b6414f085f9e08d6b3d8b585%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636894140675209934&sdata=2hphwGnJIbON1Ke9kkCIoTPg9OMs4ObiKS8SbAWIXhQ%3D&reserved=0>:

> @@ -257,8 +258,19 @@ function renderSnippets() {

     });

 }



+function responsiveResize(){

+    // Add Responsive Adjustments depending on screen size

+    $( window ).resize(function() {



I was checking that Semantic UI depends on jquery 😟

I found this https://github.com/morganbarrett/Vanilla-Semantic-UI<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmorganbarrett%2FVanilla-Semantic-UI&data=02%7C01%7Cjhalleux%40microsoft.com%7Ca8d6f4d9b6414f085f9e08d6b3d8b585%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636894140675219942&sdata=j3nN7ZL7Z3S5%2BdoIm3M89uKcYean1oefocWzcLRWcpM%3D&reserved=0> that tries to remove that dependency.

I think in this case (docs) we would need to talk more about this. What do you think?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2Fpxt%2Fpull%2F5313%23discussion_r270241070&data=02%7C01%7Cjhalleux%40microsoft.com%7Ca8d6f4d9b6414f085f9e08d6b3d8b585%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636894140675219942&sdata=yLO9aJzQg%2FGunOJ9TJF5rit8JwLyqLpuGwLnlyr3F5k%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAD-4KZYbrCJxqvkdrIJIHViMc8ciinHpks5vbVYxgaJpZM4be8sH&data=02%7C01%7Cjhalleux%40microsoft.com%7Ca8d6f4d9b6414f085f9e08d6b3d8b585%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636894140675229951&sdata=92JOhdAfHSBDx3A3BDSa64kN9dEN4CFqT4B0%2BYKs2FE%3D&reserved=0>.

pelikhan commented on docfiles/docs.js:
```diff
@@ -261,7 +261,7 @@ function renderSnippets() {
 function responsiveResize(){
     // Add Responsive Adjustments depending on screen size 
-    $( window ).resize(function() {
+    window.addEventListener('resize', evt => {
```
Watch out, you are in JavaScript like so you cannot use arrow functions, they are not supported in IE11. In general, testing your changes in IE11 is going to be the ultimate test.

toledoal replied:
I totally forgot about good old IE11

pelikhan replied:
Still popular in schools.

pelikhan commented on docfiles/footer.html:
```diff
@@ -24,8 +8,9 @@
-            <a class="ui centered item" href="https://makecode.com/" title="Microsoft MakeCode" target="_blank" rel="noopener">Powered by Microsoft MakeCode</a>
+        <div class="ui container horizontal small link list poweredBy">
+            <span class="item">Powered by</span> 
+            <a class="item middle aligned" href="https://makecode.com/" title="Microsoft MakeCode" target="_blank" rel="noopener"> <img width="100" src="/static/Microsoft-logo_rgb_c-gray.png" alt=""></a> 
```
add alt description

pelikhan commented:
Aside for IE issues, this PR looks good to me.

toledoal commented:
I'll be adding changes to the Icons of the docs ```Blocks``` content. Please check it out. And question should I continue using this PR or should I open a new one. Thanks! @pelikhan @abchatra @riknoll

pelikhan commented:
Screenshots?

toledoal commented:
We are basically moving the icons to the top right.

![image](https://user-images.githubusercontent.com/704682/55839759-a5176180-5aee-11e9-9531-7e89d2d281d0.png)

And adding some hover effect

![image](https://user-images.githubusercontent.com/704682/55839794-c5472080-5aee-11e9-9af1-0bc9febc82b8.png)

toledoal commented:
Adding the accordion in root level
![image](https://user-images.githubusercontent.com/704682/56294028-d8aa4b00-60ef-11e9-8087-de7b90b4fc7b.png)

pelikhan commented on pxtlib/docsrender.ts:
```diff
@@ -231,7 +231,7 @@ namespace pxt.docs {
 
         params["menu"] = (theme.docMenu || []).map(e => recMenu(e, 0)).join("\n")
-        params["TOC"] = TOC.map(e => recTOC(e, 0)).join("\n")
+        params["TOC"] = TOC.map(e => recTOC(e, 1)).join("\n")
```
changes in render require cloud redeploy

toledoal replied:
ok. this applies also for anything living in pxtlib right? I'm working on the lang picker for docs, which will have changes there.

pelikhan replied:
render is special because it is also used by our cloud to render doc pages. Otherwise changes in pxtlib mostly impact the client.

pelikhan replied:
This means that we need to coordinate upgrading the cloud and merging this PR

toledoal commented:
I removed the changes to docsrenderer.ts and added the styles for the "things to do" elements.
We have now a Hero macro for main or hub pages of Courses, Projects, Lessons.
The Docs now display the docMenu we have in pxtarget.json to navigate across categories of content

![demo1](https://user-images.githubusercontent.com/704682/59699044-7d134100-91b6-11e9-83aa-ff69c1e3c429.gif)

Animation and responsiveness added.

![demo3](https://user-images.githubusercontent.com/704682/59699066-856b7c00-91b6-11e9-83b3-de002c10e227.gif)

pelikhan commented on pxtlib/docsrender.ts:
```diff
@@ -294,6 +294,7 @@ namespace pxt.docs {
                 <i class="content icon"></i>
             </a>
         `
+
```
remove

pelikhan commented on pxtrunner/renderer.ts:
```diff
@@ -41,7 +41,7 @@ namespace pxt.runner {
 
     function appendBlocks($parent: JQuery, $svg: JQuery) {
-        $parent.append($('<div class="ui content blocks"/>').append($svg));
+        $parent.append($('<div class="ui content blocks"><div class="right">BLOCKS</div></div>').append($svg));
```
use lf("blocks") + css class to capitalize

toledoal replied:
ok

pelikhan commented on pxtrunner/renderer.ts:
```diff
@@ -55,12 +55,12 @@ namespace pxt.runner {
 
     function appendJs($parent: JQuery, $js: JQuery, woptions: WidgetOptions) {
-        $parent.append($('<div class="ui content js"><div><i class="ui icon xicon js"/>JavaScript</div></div>').append($js));
+        $parent.append($('<div class="ui content js"><div class="right">JAVASCRIPT</div></div>').append($js));
```
no translation needed here.

pelikhan commented on theme/themes/pxt/globals/site.variables:
```diff
@@ -36,10 +39,10 @@
 
 @docsHeaderFont: "Helvetica Neue", Helvetica, Arial, sans-serif;
-@docsPageFont: "Helvetica Neue", Helvetica, Arial, sans-serif;
+@docsPageFont: "Times New Roman", serif;
```
really "Times New Roman"?

toledoal replied:
It is mainly for descriptions in card. It doesn't look too bad. We can discuss as well.

![image](https://user-images.githubusercontent.com/704682/59700690-7c2fde80-91b9-11e9-8d2e-a2ce8696a7cb.png)

toledoal replied:
(I talked to Arturo) We will switch to Helvetica, Arial

pelikhan commented:
* why do we have multiple versions of python/javascript icons? why do we need new icons? 
* are the python/jvaascript icon used in the editor as well?

pelikhan commented:
```
[run] node C:\gh\pxt\node_modules\less\bin\lessc theme/style.less built/web/semantic.css --include-path=node_modules/semantic-ui-less:node_modules/pxt-core/theme:theme/foo/bar --no-ie-compat
Skipped data-uri embedding of ../docs/static/logo/icons/python-v2.svg because file not found
Skipped data-uri embedding of ../docs/static/logo/icons/python-v2.svg because file not found
Skipped data-uri embedding of ../docs/static/logo/icons/python-v2.svg because file not found
Skipped data-uri embedding of ../docs/static/logo/icons/python-v1.svg because file not found
```

pelikhan commented:
I'm getting a double scroll bar.
![image](https://user-images.githubusercontent.com/4175913/59700899-3ff48200-91a9-11e9-9de4-133438132378.png)

pelikhan commented:
when clicking on the links in the menu (i.e. "JavaScript"), it jumps to that page but scrolls to the bottom.

toledoal commented:
Yes, I'm missing the python logo. I'll add those svg's. These new logos (js, python) are used in the block box here (the python when using python of course):

![image](https://user-images.githubusercontent.com/704682/59701030-4a6b4780-91ba-11e9-8edf-fe60e23aa814.png)

pelikhan commented:
Print CSS needs attention. Lots of blank area in cards since background-image does not get printed: 
![image](https://user-images.githubusercontent.com/4175913/59701134-c01ae780-91a9-11e9-9a82-73ed28752ea6.png)

pelikhan commented:
We already have a python logo in the semantic icon font.

pelikhan commented:
Looks pretty slick otherwise

toledoal commented:
For the Print CSS issues. Is it ok if I move the changes from the other PR here? And remove the docsrender.ts change

pelikhan commented:
We can fix the print later yes

pelikhan commented:
Capitalizing the API names makes them less readable -- it looks like a rendering bug.
![image](https://user-images.githubusercontent.com/4175913/59701339-4800f180-91aa-11e9-8795-96c286f02d22.png)

I think they are too big. These are the JavaScript names and they are not localized.

toledoal commented:
For the API Name capitalization (got it). It has to do with the Styles applied to the ui.card. I'll add a different style when we are using "blocklyPreview" elements. I guess this is when we use this type of content title?

pelikhan commented on docfiles/style.css:
```diff
@@ -368,12 +379,21 @@ svg {
     right: 0;
     left: 0;
     bottom: 4rem;
+<<<<<<< HEAD
```
Oop

toledoal replied:
Fix in a bit

toledoal commented:
Pending Items (June 24) for Mobile

- [ ] Correct Placement of search input on Mobile
- [ ] Correct Placement and interaction of Content Menu in Mobile
- [ ] Use overlay with sidebar when in Mobile
- [ ] Sidebar height in iPhone 100%
- [ ] Search box Width in Mobile


Pending Items (June 24) for Desktop

- [ ] Hide Sticky list content when Sidebar is opened
- [ ] Fix content-width in iPad width
- [ ] Remove resize windows method (no need anymore)
- [ ] Clean up code docs.js
- [ ] Clean up code style.css (move to docs.less)

toledoal commented:
I added mobile behaviors for the sidebar and the sticky content. Basically we are hiding/showing buttons and displaying different styles.

![demo_mobile](https://user-images.githubusercontent.com/704682/60055362-fcb37b00-96a2-11e9-9d51-1ffa67d15a59.gif)

pelikhan commented on docfiles/docs.js:
```diff
@@ -477,7 +477,7 @@ function setStickyColumn() {
                 }, 1000);
                 if (window.navigator.userAgent.indexOf('Edge') !== -1) {
-                    heading.scrollIntoView();
+                    heading.nextElementSibling.scrollIntoView(false);
```
You could test that nextElemenySibling exists and scroll into view too

toledoal replied:
I found a better solution adding a few Edge CSS specific styles :)

pelikhan commented:
@riknoll @abchatra this looks good to go for me.

abchatra commented:
I reverted as this didn't work
